### PR TITLE
fix(macos/#1462): fix macOS titlebar conditional rendering

### PR DESCRIPTION
**Pull request by zbaylin**
Before the titlebar was hidden when the window was maximized. Now it's hidden when the window is fullscreen!

image

### CODE REVIEW

File: src/Store/Reducer.re [hunk 1]
```diff
@@ -53,8 +53,13 @@ let reduce: (State.t, Actions.t) => State.t =
       | WindowFocusGained => {...s, windowIsFocused: true}
       | WindowFocusLost => {...s, windowIsFocused: false}
       | WindowMaximized => {...s, windowIsMaximized: true}
```
Comment (bryphe) [inline src/Store/Reducer.re:55]: Is it possible to go from `fullscreen` to `maximized`? If so, the `windowIsFullScreen` value might stay `true` when the `WindowMaximized` event comes in. I couldn't make this happen, though. Maybe we should set `windowIsFullScreen: false` in the `WindowMaximized` case, though?

Comment (zbaylin) [reply src/Store/Reducer.re:55]: That's a great question -- I didn't consider that! I don't think it's possible, as I couldn't get it to do that either. The second point is correct though -- I'll add that now!

Comment (zbaylin) [reply src/Store/Reducer.re]: This is fixed in the latest commit! Thanks again, @bryphe!

File: src/Store/Reducer.re [hunk 2]
```diff
@@ -52,14 +52,10 @@ let reduce: (State.t, Actions.t) => State.t =
       | ReallyQuitting => {...s, isQuitting: true}
       | WindowFocusGained => {...s, windowIsFocused: true}
       | WindowFocusLost => {...s, windowIsFocused: false}
-      | WindowMaximized => {...s, windowIsMaximized: true}
-      | WindowFullscreen => {...s, windowIsFullscreen: true}
+      | WindowMaximized => {...s, windowDisplayMode: Maximized}
```
Comment (bryphe) [inline src/Store/Reducer.re:55]: Thanks for moving from the correlated-bools to a variant, this seems much easier to work with than trying to understand the relationship between `isMaximized` and `isFullscreen` - now those states are mutually exclusive 👍

**Pull request merged**